### PR TITLE
DeepReadonly support for the 'unknown' type

### PR DESCRIPTION
**Pull request by itayronen**
Fixes: https://github.com/krzkaczor/ts-essentials/issues/197

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -156,6 +156,8 @@ export type DeepReadonly<T> = T extends Builtin
   ? Promise<DeepReadonly<U>>
   : T extends {}
   ? { readonly [K in keyof T]: DeepReadonly<T[K]> }
+  : T extends unknown
```
Comment (itayronen) [inline lib/types.ts:159]:
```suggestion
  : T extends unknown
```

File: lib/types.ts [hunk 2]
```diff
@@ -156,6 +156,8 @@ export type DeepReadonly<T> = T extends Builtin
...
+  ? readonly unknown
```
Comment (itayronen) [inline lib/types.ts:160]:
```suggestion
  ? unknown
```

**Comment by Beraliv** [comment]
@itayronen thank you for the PR! Can you also please add test cases?

**Comment by itayronen** [comment]
@Beraliv I pushed a test

**Comment by Beraliv** [comment]
LGTM

@krzkaczor WDYT?

**Review by Beraliv — APPROVED** [review]

**Comment by Beraliv** [comment]
@itayronen please run prettier against the changed files

**Comment by itayronen** [comment]
Oh I need to run the `test` specifically.
Maybe add the tests folder to the `build` or `formatDeclarations` 😅  many scripts for a tiny project in my opinion.

**Comment by Beraliv** [comment]
@itayronen yeah, you need to run tests locally to format it automatically

**Comment by Beraliv** [comment]
Now everything looks good

File: lib/types.ts [hunk 3]
```diff
@@ -156,6 +156,8 @@ export type DeepReadonly<T> = T extends Builtin
   ? Promise<DeepReadonly<U>>
   : T extends {}
   ? { readonly [K in keyof T]: DeepReadonly<T[K]> }
+  : T extends unknown
```
Comment (Beraliv) [inline lib/types.ts:159]: Also, instead of `T extends unknown` just in case, let's use `unknown extends T`

Example here – https://tsplay.dev/Nd4vXN

**Comment by Beraliv** [comment]
@all-contributors please add @itayronen for code

**Pull request merged**